drupal-8/README.md

```diff
@@ -0,0 +1,155 @@
+
+```
+SOURCE_DRUPAL=$HOME/Projects/18f/cf-ex-drupal/drupal-8/
+cp -r $SOURCE_DRUPAL/.bp-config .
+git add .bp-config
+for f in .cfignore .gitignore apt.yml bootstrap.sh manifest.yml web/sites/default/settings.cf.php; do
+  cp $SOURCE_DRUPAL/$f $f
+  git add $f
+done
+```
+
+Add the service parsing to `settings.php` by pasting in the following:
+```
+if (file_exists($app_root . '/' . $site_path . '/settings.cf.php')) {
+  include $app_root . '/' . $site_path . '/settings.cf.php';
+}
+```
+
+Now you can push as you did above:
+
+
+# Debugging
+
+## SSH Setup
+
+If you `cf ssh` to the application, you'll need to set the following variables to work effectively with the DB or PHP:
+
+```
+export HOME=/home/vcap/app
+export DEPS_DIR=/home/vcap/deps
+export PYTHONPATH=/home/vcap/app/.bp/lib
+export TMPDIR=/home/vcap/tmp
+export LIBRARY_PATH=/home/vcap/deps/0/lib
+export PHPRC=/home/vcap/app/php/etc
+export PHP_INI_SCAN_DIR=/home/vcap/app/php/etc/php.ini.d
+export LD_LIBRARY_PATH=/home/vcap/deps/0/lib:/home/vcap/app/php/lib
+export PATH=/home/vcap/deps/0/bin:/usr/local/bin:/usr/bin:/bin:/home/vcap/app/php/bin:/home/vcap/app/php/sbin
+```
+
+## Development notes
+
+Use a dedicated mysql DB so it's easier to clean up without having to reprovision:
+
+
+# References
+
+These are complementary to what's described here for running on Heroku
+
+https://www.fomfus.com/articles/how-to-create-a-drupal-8-project-for-heroku-part-1
+https://www.fomfus.com/articles/how-to-deploy-a-drupal-8-project-to-heroku-part-2
+
+
+# Known issues
+
+- [x] Flysystem s3 needs testing
+- [x] HASH SALT not set yet
+- [ ] Install with standard profile instead of minimal
+- [ ] Needs testing in terms of fresh install from composer
+- [x] Determine if `apt` buildpack is still necessary with `drupal-console`, as it may use PHP libraries instead of the mysql command line. 
+  - ~Answer: not needed~ Still need mysql for `drush`, even though it's not needed for `drupal site-install`
+- [ ] Drupal install requires more than 256MB; need to drop memory limit from 512MB on install back to default 128MB.
+- [ ] Install needs manual intervention to enable S3 storage
+- [x] Install has Error: "Your sites/default/settings.php file must define the $config_directories variable as an array containing the names of directories in which configuration files can be found. It must contain a sync key."
+- [ ] Install has Error: "The trusted_host_patterns setting is not configured in settings.php. This can lead to security vulnerabilities. It is highly recommended that you configure this. See Protecting against HTTP HOST Header attacks for more information."
+- [ ] Install has Warning: "PHP OPcode caching can improve your site's performance considerably. It is highly recommended to have OPcache installed on your server."
+- [ ] Handling of ADMIN_PASS: Change to require the env var setting, or fail; or write to local FS and get via `cf ssh`
```

Comment:
I'd require the env var. We don't want people to ever have to use SSH if they can help it; in fact it should be disabled for production deployments.